jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java

```diff
@@ -479,18 +484,26 @@ public JaegerSpan startManual() {
     private ScopeManager scopeManager = new ThreadLocalScopeManager();
     private BaggageRestrictionManager baggageRestrictionManager = new DefaultBaggageRestrictionManager();
     private boolean expandExceptionLogs;
+    private JaegerObjectFactory objectFactory;
 
-    public Builder(String serviceName) {
+    public Builder(String serviceName, JaegerObjectFactory objectFactory) {
       this.serviceName = checkValidServiceName(serviceName);
-      TextMapCodec textMapCodec = new TextMapCodec(false);
+      this.objectFactory = objectFactory;
+
+      TextMapCodec textMapCodec = TextMapCodec.builder().withUrlEncoding(false).withTracingFactory(this.objectFactory).build();
       this.registerInjector(Format.Builtin.TEXT_MAP, textMapCodec);
       this.registerExtractor(Format.Builtin.TEXT_MAP, textMapCodec);
-      TextMapCodec httpCodec = new TextMapCodec(true);
+      TextMapCodec httpCodec = TextMapCodec.builder().withUrlEncoding(true).withTracingFactory(this.objectFactory).build();
       this.registerInjector(Format.Builtin.HTTP_HEADERS, httpCodec);
       this.registerExtractor(Format.Builtin.HTTP_HEADERS, httpCodec);
       // TODO binary codec not implemented
     }
 
+    @Deprecated
+    public Builder(String serviceName) {
+      this(serviceName, new JaegerObjectFactory());
+    }
```

Comment:
again, internal package, do we need this?

Reply:
Nope.